pkg/scheduler/scheduler.go

```diff
@@ -822,3 +856,32 @@ func (s *Scheduler) allClustersInReadyState(tcs []workv1alpha1.TargetCluster) bo
 	}
 	return true
 }
+
+func (s *Scheduler) reconcileEstimatorConnection(key util.QueueKey) error {
+	name, ok := key.(string)
+	if !ok {
+		return fmt.Errorf("failed to reconcile estimator connection as invalid key: %v", key)
+	}
+
+	_, err := s.clusterLister.Get(name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			s.schedulerEstimatorCache.DeleteCluster(name)
+			return nil
+		}
+		return err
+	}
+	return estimatorclient.EstablishConnection(name, s.schedulerEstimatorCache, s.schedulerEstimatorPort)
+}
```

Comment:
This reconciling guarantees we have a connection to the cluster, does this connection always available in case of the `karmada-scheduler-estimator` restart?

Reply:
No, but it doesn't matter. The gRPC library will automaticlly keep the connection alive. If the `karmada-scheduler-estimator` restarts, the gRPC connection will close and retry connecting until connection established.